library/oneview_enclosure_group_facts.py

```diff
@@ -115,31 +115,24 @@ class EnclosureGroupFactsModule(OneViewModuleBase):
 
     def __init__(self):
         super(EnclosureGroupFactsModule, self).__init__(additional_arg_spec=self.argument_spec)
+        self.set_resource_object(self.oneview_client.enclosure_groups)
 
     def execute_module(self):
         facts = {}
-        name = self.module.params.get('name')
+        enclosure_groups = []
+        name = self.module.params.get("name")
 
         if name:
-            enclosure_groups = self.oneview_client.enclosure_groups.get_by('name', name)
-
-            if enclosure_groups and "configuration_script" in self.options:
-                facts["enclosure_group_script"] = self.__get_script(enclosure_groups)
+            if self.current_resource:
+                enclosure_groups = self.current_resource.data
+                if "configuration_script" in self.options:
+                    facts["enclosure_group_script"] = self.current_resource.get_script()
         else:
-            enclosure_groups = self.oneview_client.enclosure_groups.get_all(**self.facts_params)
+            enclosure_groups = self.resource_client.get_all(**self.facts_params)
 
         facts["enclosure_groups"] = enclosure_groups
         return dict(changed=False, ansible_facts=facts)
 
-    def __get_script(self, enclosure_groups):
```

Comment:
Hope this is supported in 600 and 800 (c7000) versions.

Reply:
Yes, we haven't deleted this method, just removed the unwanted lines. We are handling this method call at line 129.